test_functions.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+from functions import get_pages, generate_pagination_links, get_countries
+
+class test_functions(unittest.TestCase):
+    # Test Suite #
+    def test_get_pages(self):
+        # Test if function returns the correct number of pages #
+        
+        self.assertEqual(get_pages(20,5), 4)
+        self.assertEqual(get_pages(21,5), 5)
+        self.assertEqual(get_pages(19,4), 5)
+        assert type(get_pages(19,4)) is int
+
+        
+    def test_get_pagination_links(self):
+        # Test if function returns an array of urls for pagination buttons #
+
+        self.assertEqual(generate_pagination_links(0, 10, 3, 'recipes', 'null', 'Paul'),
+        ['/Paul/recipes?limit=10&offset=0','/Paul/recipes?limit=10&offset=10','/Paul/recipes?limit=10&offset=20','/Paul/recipes?limit=10&offset=30'])
+        
+        self.assertEqual(generate_pagination_links(0, 10, 3, 'search', 'chicken', 'Paul'),
+        ['/Paul/search/chicken?limit=10&offset=0','/Paul/search/chicken?limit=10&offset=10',
+        '/Paul/search/chicken?limit=10&offset=20','/Paul/search/chicken?limit=10&offset=30'])
+        
+        self.assertEqual(generate_pagination_links(0, 10, 4, 'search', 'chicken', 'Jim'),
+        ['/Jim/search/chicken?limit=10&offset=0','/Jim/search/chicken?limit=10&offset=10',
+        '/Jim/search/chicken?limit=10&offset=20','/Jim/search/chicken?limit=10&offset=30','/Jim/search/chicken?limit=10&offset=40'])
+        
+        assert type(generate_pagination_links(0, 10, 3, 'recipes', 'null','Kim')) is list
+    
+    def test_get_counties(self):
+        # Test if function returns an array of countries  #
+        
```

Comment:
Read the first/last object and assert any of its content.

Reply:
This has been updated